packages/nostr_sdk/lib/relay/relay.dart

```diff
@@ -75,9 +75,57 @@ abstract class Relay {
 
   Future<void> disconnect();
 
-  bool _waitingReconnect = false;
+  /// Whether we are waiting to reconnect to this relay (due to exponential
+  /// backoff from a previous error)
+  @visibleForTesting
+  bool waitingReconnect = false;
 
-  void onError(String errMsg, {bool reconnect = false}) {
+  /// The number of times we've attempted to reconnect to this relay since the
+  /// last successful connection.
+  @visibleForTesting
+  int reconnectAttempts = 0;
+
+  /// The maximum time we will wait before attempting to reconnect to
+  /// this relay.
+  static const int maxDelaySeconds = 32;
+
+  /// The base delay in seconds before attempting to reconnect after an error
+  final int _reconnectBaseDelay = 1;
+
+  /// Reset reconnect attempt counter
+  void resetReconnectAttempts() {
+    reconnectAttempts = 0;
+  }
+
+  /// Calculate delay using exponential backoff with jitter
+  @protected
+  Duration calculateReconnectDelay() {
+    // First attempt: reconnect immediately
+    if (reconnectAttempts <= 1) return Duration.zero;
+
+    // Second attempt: wait just 1 second
+    if (reconnectAttempts == 2) return const Duration(seconds: 1);
+
+    // For subsequent attempts, use exponential backoff
+    // Exponential backoff: baseDelay * 2^(attempt-2)
+    // We subtract 2 from the attempt count since we're starting exponential
+    // backoff from the third attempt onward.
+    final int adjustedAttempt = reconnectAttempts - 2;
+    final double backoffFactor = (1 << adjustedAttempt).toDouble();
+    final int delaySeconds = (_reconnectBaseDelay * backoffFactor).round();
+    final int cappedDelaySeconds =
+        delaySeconds > maxDelaySeconds ? maxDelaySeconds : delaySeconds;
+
+    // Add jitter (±10% variation) to prevent reconnection storms
+    final random =
+        (DateTime.now().microsecondsSinceEpoch % 1000) / 10000; // 0.0-0.1
+    final jitterFactor = 0.9 + random * 0.2; // 0.9-1.1
+
+    return Duration(
+        milliseconds: (cappedDelaySeconds * 1000 * jitterFactor).round());
+  }
+
+  void onError(String errMsg, {bool reconnect = true}) {
```

Comment:
Being really picky, but I prefer for booleans to be non-imperative verb phrases, like Effective Dart describes [here](https://dart.dev/effective-dart/design#prefer-a-non-imperative-verb-phrase-for-a-boolean-property-or-variable). It slightly increases readability, lowering cognitive load.

```suggestion
  void onError(String errMsg, {bool shouldReconnect = true}) {
```

Reply:
Good suggestions (I didn't write this function, despite the fact that it shows up in the diff, but I'm happy to refactor.)